datagen/dataset_pzipfx.go

```diff
@@ -0,0 +1,282 @@
+	parType  PartType
+}
+
+var tables = []tableMeta{
+	{
+		name:     "p-hash-zint-a",
+		dataType: TypeInt,
+		parCol:   "a",
+		parType:  HashPart,
+	},
+	{
+		name:     "p-hash-zint-c",
+		dataType: TypeInt,
+		parCol:   "c",
+		parType:  HashPart,
+	},
+	{
+		name:     "p-range-zint-a",
+		dataType: TypeInt,
+		parCol:   "a",
+		parType:  RangePart,
+	},
+	{
+		name:     "p-range-zint-c",
+		dataType: TypeInt,
+		parCol:   "c",
+		parType:  RangePart,
+	},
+	{
+		name:     "p-range-zdouble-a",
+		dataType: TypeDouble,
+		parCol:   "a",
+		parType:  RangePart,
+	},
+	{
+		name:     "p-range-zdouble-c",
+		dataType: TypeDouble,
+		parCol:   "c",
+		parType:  RangePart,
+	},
+	{
+		name:     "p-range-zstring-a",
+		dataType: TypeString,
+		parCol:   "a",
+		parType:  RangePart,
+	},
+	{
+		name:     "p-range-zstring-c",
+		dataType: TypeString,
+		parCol:   "c",
+		parType:  RangePart,
+	},
+	{
+		name:     "p-range-zdatetime-a",
+		dataType: TypeDateTime,
+		parCol:   "a",
+		parType:  RangePart,
+	},
+	{
+		name:     "p-range-zdatetime-c",
+		dataType: TypeDateTime,
+		parCol:   "c",
+		parType:  RangePart,
+	},
+}
+
+type pZipfXOpt struct {
+	x            float64
+	n            int64
+	ndv          int64
+	partitionNum int64
+}
+
+func parsePZipfXOpt(args string) (opt pZipfXOpt, err error) {
+	kvs := strings.Split(args, ",")
+	for _, kv := range kvs {
+		tmp := strings.Split(kv, "=")
+		if len(tmp) != 2 {
+			return opt, errors.Errorf("invalide kv=%v", kv)
+		}
+		k, v := tmp[0], tmp[1]
+		switch strings.ToLower(k) {
+		case "x":
+			if opt.x, err = strconv.ParseFloat(v, 64); err != nil {
+				return opt, errors.Errorf("invalid x=%v", v)
+			}
+		case "n":
+			if opt.n, err = strconv.ParseInt(v, 10, 64); err != nil {
+				return opt, errors.Errorf("invalid n=%v", v)
+			}
+		case "ndv":
+			if opt.ndv, err = strconv.ParseInt(v, 10, 64); err != nil {
+				return opt, errors.Errorf("invalid ndv=%v", v)
+			}
+		case "partition_num":
+			if opt.partitionNum, err = strconv.ParseInt(v, 10, 64); err != nil {
+				return opt, errors.Errorf("invalid partition_num=%v", v)
+			}
+		}
+	}
+	return
+}
+
+func genPartitionSchema(t tableMeta, opt pZipfXOpt, ints []int, doubles []float64) (string, error) {
+
+	content := fmt.Sprintf("CREATE TABLE %s (a %s, b %s, c %s, key(a), key(a, b), key(b), key(b, c))\n",
+		t.name,
+		t.dataType,
+		t.dataType,
+		t.dataType)
+
+	switch t.parType {
+	case HashPart:
+		content = fmt.Sprintf("%s PARTITION BY HASH(%s) PARTITION %d;", content, t.parCol, opt.partitionNum)
+	case RangePart:
+		content = fmt.Sprintf("%s PARTITION BY RANGE(%s) (\n", content, t.parCol)
+
+		begin, _ := time.Parse(timeLayout, "2000-01-01 00:00:00")
+		end, _ := time.Parse(timeLayout, "2100-01-01 00:00:00")
+		interval := end.Sub(begin)
+
+		var i int64
+		for i = 0; i < opt.partitionNum-1; i++ {
+			var v string
+			switch t.dataType {
+			case TypeInt:
+				v = strconv.FormatUint(uint64(ints[int64(len(ints))*i/opt.partitionNum]), 10)
```

Comment:
Please use a more random method to generate the partition range. For example, if the data range belongs to [0, 10000]. We can use 10000/partitionNum to generate the partition range.

Reply:
I'd like to merge this PR and solve this problem later.